biz.aQute.bndlib/src/aQute/bnd/cdi/CDIAnnotations.java

```diff
@@ -304,4 +304,28 @@ private Document readXMLResource(Resource resource) {
 		}
 	}
 
+	static class Tuple {
+		final String	resourcePath;
+		final Resource	resource;
+		final Discover	discover;
+
+		public Tuple(String resourcePath, Resource resource, Discover discover) {
+			this.resourcePath = resourcePath;
+			this.resource = resource;
+			this.discover = discover;
+		}
+
+		boolean sameOrigin(Clazz c) {
+			if (!resource.getClass()
+				.equals(c.getResource()
+					.getClass())) {
+				return false;
+			}
+			if (resourcePath.contains(c.getClassName()
```

Comment:
The contains test seems too weak. It would seem that `com/foo/bar/Outer$StaticInner` contains `com/foo/bar/Outer` but they are not the same class.